notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb

```diff
@@ -0,0 +1,853 @@
+   "metadata": {},
+   "source": [
+    "# Step 8: Get Row Count of Tables and identify Partition Columns\n",
+    "\n",
+    "This step uses PARTITION_THRESHOLD (default value is 1 million) parameter and any table having rows greater than PARTITION_THRESHOLD will be partitioned based on Primary Keys\n",
+    "Get Primary keys for all tables to be migrated and find an integer column to partition on"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "636b871e-c975-4a0b-bc51-7069ce77e51e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PARTITION_THRESHOLD = 1000000\n",
+    "CHECK_PARTITION_COLUMN_LIST={}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f2055755-60ad-4371-a936-bfbfa8aaedb8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with DB.connect() as conn:\n",
+    "    for table in SQL_SERVER_TABLE_LIST:\n",
+    "        results = DB.execute(\"SELECT count(1) FROM {}\".format(table)).fetchall()\n",
+    "        # print(results)\n",
+    "        if results[0][0]>PARTITION_THRESHOLD and len(SQL_TABLE_PRIMARY_KEYS.get(table).split(\",\")[0])>0:\n",
+    "            column_list=SQL_TABLE_PRIMARY_KEYS.get(table).split(\",\")\n",
+    "            for column in column_list:\n",
+    "                results_datatype = DB.execute(\"SELECT DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = '{0}' AND TABLE_NAME   = '{1}' AND COLUMN_NAME  = '{2}'\".format(table.split(\".\")[0],table.split(\".\")[1],column)).fetchall()      \n",
+    "                # print(results_datatype)\n",
+    "                if results_datatype[0][0]==\"int\":\n",
+    "                    lowerbound = DB.execute(\"SELECT min({0}) from {1}\".format(column,table)).fetchall()\n",
+    "                    upperbound = DB.execute(\"SELECT max({0}) from {1}\".format(column,table)).fetchall()\n",
+    "                    numberPartitions = math.ceil((upperbound[0][0]-lowerbound[0][0])/PARTITION_THRESHOLD)\n",
+    "                    CHECK_PARTITION_COLUMN_LIST[table]=[column,lowerbound[0][0],upperbound[0][0],numberPartitions]\n",
+    "                    print(CHECK_PARTITION_COLUMN_LIST)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ff7d7523-4f87-48e7-bbe5-d1206b817c3d",
+   "metadata": {},
+   "source": [
+    "# Step 9: Execute Pipeline to Migrate tables from MSSQL to BIGQUERY\n",
+    "\n",
+    "* BIGQUERY_DATASET : Target dataset in Bigquery\n",
+    "* BIGQUERY_MODE : Mode of operation at target <append|overwrite|ignore|errorifexists> (default overwrite)\n",
+    "* TEMP_GCS_BUCKET : Bucket name for dataproc job staging\n",
+    "* PYTHON_FILE_URIS : Path to PACKAGE_EGG_FILE\n",
+    "* MAIN_PYTHON_CLASS : Path to main.py"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "46e607f2-c6c0-4abe-b4ff-ce34c83fcbf2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mssql_to_bigquery_jobs = []\n",
```

Comment:
can we move this to top in step 3?

Reply:
Moved this to step 3 where common parameters for the notebooks have been defined.